src/liballoc/rc.rs

```diff
@@ -1708,9 +1702,18 @@ impl<T> Weak<T> {
     /// [`null`]: ../../std/ptr/fn.null.html
     #[stable(feature = "weak_into_raw", since = "1.45.0")]
     pub fn as_ptr(&self) -> *const T {
-        let offset = data_offset_sized::<T>();
-        let ptr = self.ptr.cast::<u8>().as_ptr().wrapping_offset(offset);
-        ptr as *const T
+        let ptr: *mut RcBox<T> = NonNull::as_ptr(self.ptr);
+
+        // SAFETY: we must offset the pointer manually, and said pointer may be
+        // a dangling weak (usize::MAX) if T is sized. data_offset is safe to call,
```

Comment:
Should the doc comment in `pub struct Weak<T: ?Sized> ` also be updated to say that if `T` is not sized then it definitely is a dereferencable pointer? Now that we rely on this invariant, we better make sure other people editing this code are aware of it.

That is my last comment. :)

Reply:
Fair; added a line to this effect.